### PR TITLE
[teamcity] Add basic_http_authentication support

### PR DESCRIPTION
**Pull request by mderomph-coolblue**

### CODE REVIEW

**Comment by mderomph-coolblue** [comment]
Bump, anybody that could merge this?

File: conf.d/teamcity.yaml.example [hunk 1]
```diff
@@ -10,10 +10,14 @@ instances:
     # Specify the server name of your teamcity instance here
     # Guest authentication must be on if you want the check to be able to get data
     server: teamcity.mycompany.com
```
Comment (olivielpeau) [inline conf.d/teamcity.yaml.example]: Could you mention here that when basic auth is used the username and password need to be provided in the `server` parameter in the `user:password@server` format?

**Comment by olivielpeau** [comment]
Hi @mderomph-coolblue, thanks for your contribution!

I had a quick look at your PR and had only one comment for now, could you address it please?

We'll be sure to review your PR in details soon.

**Comment by mderomph-coolblue** [comment]
Passed all the test now, hope you guys can take a look at it. Thanks!

**Comment by mderomph-coolblue** [comment]
Is there anythings else I can do to get this pull request merged?

**Comment by remh** [comment]
Looks good to me. Thanks!

**Pull request merged**